examples/lines_bars_and_markers/gradient_bar.py

```diff
@@ -10,12 +24,44 @@
 np.random.seed(19680801)
 
 
-def gbar(ax, x, y, width=0.5, bottom=0):
-    X = [[.6, .6], [.7, .7]]
+def gradient_image(ax, extent, direction=0.3, cmap_range=(0, 1), **kwargs):
+    """
+    Draw a gradient image based on a colormap.
+
+    Parameters
+    ----------
+    ax : Axes
+        The axes to draw on.
+    extent
+        The extent of the image as (xmin, xmax, ymin, ymax).
+        By default, this is in Axes coordinates but may be
+        changed using the *transform* kwarg.
+    direction : float
+        The direction of the gradient. This is a number in
+        range 0 (=vertical) to 1 (=horizontal).
+    cmap_range : float, float
+        The fraction (cmin, cmax) of the colormap that should be
+        used for the gradient, where the complete colormap is (0, 1).
+    **kwargs
+        Other parameters are passed on to `.Axes.imshow()`.
+        In particular useful is *cmap*.
+    """
+    phi = direction * np.pi / 2
+    v = np.array([np.cos(phi), np.sin(phi)])
+    X = np.array([[v @ [1, 0], v @ [1, 1]],
+                  [v @ [0, 0], v @ [0, 1]]])
```

Comment:
Just a style/judgment thing.

I *think* this is
```
np.dot(v, [[[1, 0], [1, 1]], [[0, 0], [1, 1]]]
```
or alternatively, just don't bother with dot products at all
```
s = np.sin(phi)
c = np.cos(phi)
X = [[s, s+c], [0, c]]
```
(either fully vectorize the thing, or don't vectorize it at all.)

Reply:
I've chosen this notation intentionally to illustrate the projection of the corners on *v*.

The upper does not produce the same result (I assume it's some axis ordering, but didn't investigate). The lower does work, but obscures what the calculation actually does.

Reply:
Ah, I didn't see it as a projection; indeed, that makes things clearer.